pandas/core/indexes/base.py

```diff
@@ -2858,13 +2858,15 @@ def _intersection(self, other, sort=False):
             indexer = algos.unique1d(Index(rvals).get_indexer_non_unique(lvals)[0])
             indexer = indexer[indexer != -1]
 
-        result = other.take(indexer).unique()._values
+        result = Index(other.take(indexer))
```

Comment:
at this point we know other is already an Index

Reply:
Yes, you are right. I mixed that up with the assert below, which is a np.array, Thx